third_party/amd/lib/Dialect/TritonAMDGPU/IR/Dialect.cpp

```diff
@@ -132,6 +129,46 @@ LogicalResult ExtractSliceOp::verify() {
                        << shapePerCTATile << "]";
   }
 
+  auto llSrc = triton::gpu::toLinearLayout(srcShape, srcLayout);
+  auto llRes = triton::gpu::toLinearLayout(resShape, resultLayout);
```

Comment:
Are the checks in L91 - L130 still useful? LL bakes in shapes already right? It looks to me that we only need to additionally check we are not taking a slice with `offsets[i]` that are larger than the total number of available slides?